topics/ui/components/tool_window.md

```diff
@@ -6,50 +6,86 @@
+
+<!--### Icon
+
+Follow [tool window icons guidelines](icons_style#NNN.md) to create correct icons for tool windows.
+// TODO: publish this chapter when the guideline about icons is public too
+-->
+
+#### Feedback
+
+If the content changes and can contain different types of feedback, for example, new updates or errors, a colored badge is shown above the icon.
+
+![](toolwindow_badge.png){width=706}
+
+Do not change the icon for the tool window button when the content changes.
+
+<!--**Exception:** <control>Problems</control> and <control>Event Log</control> icons change color and show the current status.-->
+
+### Tabs
+
+Add tabs if all the information does not fit on one screen or refers to similar instances, like run sessions, history for files, find results. The tabs for entities should be closable.
+
+![](toolwindow_tabs.png){width=706}
+
+### Orientation and content
+
+#### Vertical window
 
-![](pull_requests.png){width=308}
+Vertical tool windows work better for trees, for example, Project, Structure, or Maven tool windows.
 
-Add an icon for the tool window. The icon is 13px x 13px, grey and monochromatic. See how to create icons in the [icons guidelines](icons_style.md).
+![](toolwindow_vertical.png){width=706}
 
-**Exception:** <control>Problems</control> and <control>Event Log</control> icons change color and show the current status.
+#### Horizontal window
 
-Select the tool window orientation depending on the content. Vertical tool windows work better for trees, for example, Project, Structure, or Maven tool windows. Horizontal tool windows work better for tables, wide content or master-detail panels, for example, Git Log, Terminal, or Problems.
+Horizontal tool windows work better for tables, wide content, or master-detail panels, for example, Git Log, Terminal, or Problems.
 
-Do **not** show the tool window button if the tool window is not relevant to the current project configuration. For example, do not show the Maven tool window for a project without Maven configuration files.
+![](toolwindow_horizontal.png){width=706}
 
-If a tool window has no content yet, show its button by default only if the window contains basic functionality that is likely to be used for all projects, for example, Version Control or Problems. Otherwise, do not show the tool window button by default.
+#### Visibility
 
-Add a toolbar for [frequently used actions and filters](toolbar.md#what-items-to-add-on-toolbar). Use a horizontal toolbar for vertical tool windows and a vertical toolbar for horizontal tool windows. For more details, see the [toolbar guidelines](toolbar.md).
+If a tool window has no content yet, show its button by default only if the window contains basic functionality that is likely to be used for all projects, for example, <control>Version Control</control> or <control>Problems</control>. Otherwise, hide the tool window button by default. Hidden tool windows are still available under <control>More tool windows</control> menu.
```

Comment:
Possibly add a link to the Empty state article for a tool window without content?